### DISCUSSION TITLE
Cloudflare Tunnel with internal dns server

### DISCUSSION THREAD

**Discussion by aj-bayanat** [Ylianst/MeshCentral#4599] [Q&A]
Hey,
I was wondering how to setup MC with Cloudflare tunnels. This would be ideal as I do not need to port forward. 
In the config file I have tlsoffload set to true and certurl to mesh.mydomain.com.
However when I am using Cloudflare tunnels, MC is unable to get the SHA of the cert.
mydomain.com is also used for internal routing on my windows dns server. To "bypass "my windows server and go to Cloudflare normally I add a CNAME entry like mesh.mydomain.com.cdn.cloudflare.net. 
With this any of my internal clients can access mesh.mydoin.com by going through Cloudflare (I do this for other webapps for SSO and access control). With this setup however for some reason MC is not able to grab the SHA from Cloudflare.

So basically internal clients would ask for dns of mesh.mydomain.com --receive a CNAME -- go to through Cloudflare

Any suggestions would be appreciated.
